libs/designer/src/lib/core/state/templates/manifestSlice.ts

```diff
@@ -7,6 +7,23 @@ import type { FilterObject } from '@microsoft/designer-ui';
 export const templatesCountPerPage = 25;
 const initialPageNum = 0;
 
+interface ContentInfo<T> {
+  value: T;
+  isEditable?: boolean;
+}
+
+export interface ViewTemplateDetails {
+  id: string;
+  basicsOverride?: Record<
+    string,
+    {
+      name?: ContentInfo<string>;
+      kind?: ContentInfo<string>;
+    }
+  >;
+  parametersOverride?: Record<string, ContentInfo<any>>;
```

Comment:
It may be worth to have a doc outlining the contract for different type that is supported. From what I found in template repo, if the parameter type is int, the value should be a string like "123", so assuming here we want to stick to this contract to make it work. If this is not the case, maybe call out the types we tested and supported, while we fix this.

Reply:
We should ideally fix the parameter value either here or in parameters UI, it can be done as part of impoementing the overrides